src/effects/native/gatereffect.cpp

```diff
@@ -0,0 +1,236 @@
+#include "effects/native/gatereffect.h"
+
+namespace {
+    // Gain when the gate is open, higher than 1 to still have a decent level
+    constexpr double kMaxGain        = 1.5; 
+    // Attack slope
+    constexpr double kBaseAttackInc  = 0.001;
+    // Release slope
+    constexpr double kBaseReleaseInc = 0.0005;
+    }
+
+// static
+QString GaterEffect::getId() {
+    return "org.mixxx.effects.gater";
+}
+
+// static
+EffectManifest GaterEffect::getManifest() {
+    EffectManifest manifest;
+    manifest.setId(getId());
+    manifest.setName(QObject::tr("Gater"));
+    manifest.setAuthor("The Mixxx Team");
+    manifest.setVersion("1.0");
+    manifest.setDescription("");
+
+    EffectManifestParameter* rate = manifest.addParameter();
+    rate->setId("rate");
+    rate->setName(QObject::tr("Rate"));
+    rate->setDescription(QObject::tr(""));
+    rate->setControlHint(EffectManifestParameter::ControlHint::KNOB_STEPPING);
+    rate->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    rate->setUnitsHint(EffectManifestParameter::UnitsHint::UNKNOWN);
+    rate->appendStep(QPair<QString, double>("1", 0));
+    rate->appendStep(QPair<QString, double>("1/2", 0.34));
+    rate->appendStep(QPair<QString, double>("1/4", 0.67));
+    rate->appendStep(QPair<QString, double>("1/8", 1));
+
+    EffectManifestParameter* shape = manifest.addParameter();
+    shape->setId("shape");
+    shape->setName(QObject::tr("Shape"));
```

Comment:
Maybe "Hold" would be a better name for this parameter?

Reply:
I began to change it to "Hold" after your comment, but I think that "Shape" is appropriate and will fit better when moving to the more generic Tremolo effect discussed below